plugins/autogems.cpp

```diff
@@ -210,6 +212,32 @@ void create_jobs() {
     }
 }
 
+void save_config() {
+    Json::Value& data = Persistent::get("autogems");
+    data["version"] = persist_version;
+    data["running"] = running;
+}
+
+void load_config(color_ostream &out) {
+    Json::Value& data = Persistent::get("autogems");
+    int32_t version = Json::get<int>(data, "version", 0);
+    if ( version == 0 ) {
+        // Load from historical figure.
+        // Always saves back to the property_tree,
+        // to avoid looking up the histfig on every load.
+        PersistentDataItem config = World::GetPersistentData(CONFIG_KEY);
+        running = config.isValid() && !config.ival(0);
+        save_config();
+        World::DeletePersistentData(config);
+    } else if ( version == 1 ) {
+        // Load from the property_tree.
+        running = Json::get<bool>(data, "running", false);
+    } else {
+        // Collect what we can, but warn about the unknown version.
```

Comment:
Thanks.  I clearly meant to do so, and never got around to it.  Not that I expect it to matter for this plugin, but it might for something, and this could end up being used as an example.